packages/react-tinacms-github/src/toolbar-plugins/BranchSwitcherPlugin.tsx

```diff
@@ -0,0 +1,513 @@
+/**
+
+Copyright 2019 Forestry.io Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import * as React from 'react'
+import {
+  Form,
+  Input,
+  useCMS,
+  Modal,
+  ModalPopup,
+  ModalHeader,
+  ModalBody,
+  ModalActions,
+  FormBuilder,
+  FieldsBuilder,
+} from 'tinacms'
+import { AddIcon, ChevronDownIcon, LockIcon } from '@tinacms/icons'
+import { Button } from '@tinacms/styles'
+import { Dismissible } from 'react-dismissible'
+import styled, { css } from 'styled-components'
+import { GithubClient, Branch } from '../github-client'
+import { LoadingDots } from '@tinacms/react-forms'
+
+interface BranchSwitcherProps {
+  onBranchChange?(branch: string): void
+}
+
+const BranchSwitcher = ({ onBranchChange }: BranchSwitcherProps) => {
+  const cms = useCMS()
+  const github: GithubClient = cms.api.github
+
+  const [open, setOpen] = React.useState(false)
+  const [confirmSwitchProps, setConfirmSwitchProps] = React.useState<any>()
+  const [createBranchProps, setCreateBranchProps] = React.useState<any>()
+  const [filterValue, setFilterValue] = React.useState('')
+  const selectListRef = React.useRef<HTMLElement>()
+
+  const [branchStatus, setBranchStatus] = React.useState<
+    'pending' | 'loaded' | 'error'
+  >('pending')
+  const [branches, setBranches] = React.useState<Branch[]>([])
+  React.useEffect(() => {
+    github
+      .getBranchList()
+      .then(branches => {
+        setBranches(branches)
+        setBranchStatus('loaded')
+      })
+      .catch(() => {
+        setBranchStatus('error')
```

Comment:
It doesn't look like we're handling an error branchStatus currently

Reply:
Indeed we are not.

@spbyrne want to add a little "We had trouble loading branches. Please refresh to try again."

Reply:
Yeah good call.

Reply:
Added 👍